core/dbt/events/adapter_endpoint.py

```diff
@@ -13,7 +13,7 @@ class AdapterLogger:
     name: str
 
     def debug(self, msg, *args, exc_info=None, extra=None, stack_info=False):
-        event = AdapterEventDebug(name=self.name, base_msg=msg, args=args)
+        event = AdapterEventDebug(name=self.name, base_msg=str(msg), args=args)
```

Comment:
Okay, but like Mila, why not change AdapterEventBase. I tried, but what this is force us to mutate data already encapsulated which feels wrong to my sensibilities.

However...this might not prevent the issues we have with efficiency that Nate called out? See, I can take a real crack at modifying the base class or even doing a stringification downstream in fire event. I tried all that. This seems most clean to me, but I realize now this may not  be efficient, only making sense within our current spec which maybe should change.

Thoughts, o hetairoi?

Reply:
Huh.

To clarify the "performance" bit, it is about performance, but even more importantly it's about adhering to standard Python logging semantics- i.e. lazily converting arguments to strings. This doesn't actually break that because the arguments remain untouched.

Is this what's happening in the two user reported issues linked in the ticket? Logging a non-string value as the base message?

Reply:
Another idea would be to add a post init processing to convert any non-string input to string: https://docs.python.org/3/library/dataclasses.html#post-init-processing

Reply:
After the structured logging rework and Nate's additional ticket, post_init proved a very nice paradigm for insert a quick lazy-load-friendly type conversion.